### DISCUSSION TITLE
Process and Instrument Diagrams (P&ID) ?

### DISCUSSION THREAD

**Discussion by jopetpc** [jgraph/drawio-desktop#1623] [General]
Hi, I am testing the DrawIO desktop version, to find if it is suitable for P&ID drawings.
Other tools for drawing P&ID’s, like MicroStation or AutoCAD, has no export/import possibilities for xml files, only proprietary file formats (dng, dwg). And they, and any conversion tools, are quite expensive.

I am looking for a tool to create ‘intelligent’ P&ID’s in the way that data in a xml file could be imported into a database or exported from such for updates or generation of new diagrams (not only P&ID’s, but for instance Instrument Loop drawings).

So far, I have found many goodies in DrawIO, a Process Engineering library – a fairly easy way to modify shapes, definition my own properties and building a customized library.
I have tested with properties for entering Tag- or Line Numbers (instead of using labels) and imported info of the connections between process lines, tagged equipment etc.

A challenge is when a shape (for instance a pump – or even worse; a 3 way valve) is inserted in the middle of a process line, this results in tracking problems. The line is splits and it is hard to trace from start to the end based on source/target of the new segment(s) automatically generated. It would be nice to insert (and connect) one or more shapes in the middle without introducing new DrawIO id’s for the middle segments. Or maybe have the original property holding the Line number (entered once) available as a hidden property in the new segments.

My main question is: Do JGraph Ltd / draw.io AG have any plans for a version that is optimized for a P&ID drafter ? I guess there should be a marked of interest in large engineering companies and in general in the Oil&Gas industry.

Another question to this forum is: Do anyone have experience with DrawIO and P&ID’s, when it comes to interfacing using xml/xsl - without have to develop a plugin ?

**Comment by davidjgraph** [comment]
The core team does not have plans for domain specific versions of the editor.

**Comment by jopetpc** [comment]
Thank you David - for an answer anyway.
